### DISCUSSION TITLE
Preloading uncached redirects

### DISCUSSION THREAD

**Discussion by PietM** [swup/swup#865] [Help] [answered]
Hi all! 
Since updating to Swup v4, I've noticed a lag when transitioning to a new page using the Preload Plugin. With v3, transitions would be immediate. I haven't changed my code, aside from switching out "swup:contentReplaced" for  "swup:content:replace" for DOM events. 
I miss my speedy site! 
Any guidance or feedback is greatly appreciated! 
M

**Comment by daun** [comment]
Hi 🥬 How long of a delay are you seeing? And does it occur before the transition or after? The best scenario here would be for you to provide a demo link or [minimal reproduction](https://swup.js.org/other/minimal-reproductions/) so others can help narrow this down.

**Comment by PietM** [comment]
Thank you for getting back! 

Here's two quick, simplified URLs: One running v4, the other running v3. 
Click the links in the upper lefthand corner to run through the pagination. 

- - - - 

v3:  https://a-r-p.co/SWUP/SWUP2
https://github.com/swup/swup/assets/9273660/e14fd4c0-19e9-45e3-9be1-a974df4c065e

- - - - 

v4:  https://a-r-p.co/SWUP
https://github.com/swup/swup/assets/9273660/9f5cf327-dc8f-442a-8f4e-4b922233bf85

- - - - 

You will notice v3 is much faster with page loading, but the code is the same. I've just switched out the SWUP versions. 
Why is the result so different? 

Thank you again for your attention.
 I love SWUP!

**Comment by daun** [comment, marked as answer]
This might have to do with redirects, which are no longer cached in swup 4. The links in your examples point to urls without trailing slash, like `/page1`, which then redirect to the actual urls with trailing slash, like `/page1/`. We have implemented some logic to not cache redirected requests – ideally we could only ignore temporary 302 redirects but there's no way to tell so we need to ignore permanent 301 redirects as well. Can you try and see if using the "correct" urls with trailing slashes helps here? We might look into enabling redirect caching for preload requests, or into rewriting the cache url to cache the final url after the redirect.

**Comment by PietM** [comment]
Thank you for the prompt feedback @daun!

That may have helped, but there is still an observable lag and sluggishness on initial load for each page (contrasting with the alacrity and immediacy of v3) -- even with the {preloadVisibleLinks: true} option enabled. SWUP's swift transitioning is a real asset. I'd hate to lose that. Please clear your cache and compare the links again. 

Any other thoughts, or is it worthwhile to continue with v3 until these issues are addressed in future iterations? 

Thanks again for quick help. 
The assistance and help is reassuring! 
I so appreciate it.

**Comment by daun** [comment]
In the swup v4 example, there doesn't seem to be any preloading happening at all, judging from the developer tools' network tab.

You've included two versions of Preload Plugin, an older one from your server and the current one from a CDN. Try and remove the older/second script tag and see if that restores the preloading. If that's the case, the delay you're seeing is just the preloading not happening because the old Preload Plugin version doesn't work with the new swup version.

There's `https://unpkg.com/@swup/preload-plugin@3` in the `head` tag (current version) as well as `https://a-r-p.co/SWUP/wp-content/themes/THEME/js/SwupPreloadPlugin.js?ver=1.0.0` in the `body` tag (old version, incompatible with swup 4)

**Comment by PietM** [comment]
D'AH! I forgot to disable those scripts in the functions file! 
Removing the old preload script and revising the URLS fixed the issue! 
THANK YOU again for the quick help and improvements made to SWUP. 
Saved the day!

**Reply by daun** [reply]
Wonderful! Glad it's solved.
